### PR TITLE
Parse nessus vulnerabilities for sanity

### PR DESCRIPTION
**Pull request by pburkholder**
Requires:

`pip3 install nessus-file-reader`

Creates output like:

```
python3 parse-nessus-xml.py Production_Vulnerability_scan_xxxx.nessus
File name: /Users/peterburkholder/Downloads/Production_Vulnerability_scan_xxxx.nessus
File size: 67.6 MiB

33851, Risk: Low, Plugin Name: Network daemons not managed by the package system, https://www.tenable.com/plugins/nessus/33851
	foo-production
        bar-production
        baz-production
```

Solves https://github.com/18F/cg-private/issues/112

### CODE REVIEW

**Comment by pburkholder** [comment]
I've gone over the code at https://github.com/LimberDuck/nessus-file-reader/ and didn't see anything suspicious in terms of leaking information to outside sources.  But this probably could be rewritten to not require any PIP install now that I've seen how it's done.

File: parse-nessus-xml.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+vulnids = {}
+for report_host in nfr.scan.report_hosts(root):
+    report_host_name = nfr.host.report_host_name(report_host)
+    for report_item in nfr.host.report_items(report_host):
+        plugin_id = int(nfr.plugin.report_item_value(report_item, 'pluginID'))
+        risk_factor = nfr.plugin.report_item_value(report_item, 'risk_factor')
+        plugin_name = nfr.plugin.report_item_value(report_item, 'pluginName')
+        id = '{}, Risk: {}, Plugin Name: {}, https://www.tenable.com/plugins/nessus/{}'.format(
```
Comment (bengerman13) [inline parse-nessus-xml.py]: I think an f-string would make this line a lot nicer

File: parse-nessus-xml.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                                            plugin_id, risk_factor, plugin_name, plugin_id)
+        hosts = vulnids.get(id, [])
+        hosts.append(report_host_name)
+        vulnids[id] = hosts
+
+for key in vulnids:
+    if key.find("Risk: None") == -1 :
+        print(key)
+        for site in vulnids[key]:
+            print('\t{}'.format(site))
```
Comment (bengerman13) [inline parse-nessus-xml.py]: either an f-string, or just `print('\t', site)`

**Comment by bengerman13** [comment]
re: could do it on your own without the dependency - .nessus files are just XML, we can also export them as csvs if that's easier to work with

File: parse-nessus-xml.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import nessus_file_reader as nfr
+import os
+import glob
+import traceback
+import time
+import sys
+
+
+if len(sys.argv) == 1:
```
Comment (hillaryj) [inline parse-nessus-xml.py]: I generally recommend this goes inside a `if __name__ == "__main__":` block.

File: parse-nessus-xml.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+for key in vulnids:
+    if key.find("Risk: None") == -1 :
+        affected_hosts = vulnids[key]
+        print(key)
+        if len(affected_hosts) > 4:
+            print('\t{} affected hosts found ...'.format(len(affected_hosts)))
+        else:
+            for site in affected_hosts:
+                print('\t{}'.format(site))
```
Comment (hillaryj) [inline parse-nessus-xml.py]: There should be an newline (aka empty line) at the end of the file per python style guides as a line/file delimiter - I've started running `black` on my files to have it take care of formatting per some discussion in the python or dev channels on Slack and it works really well.

**Comment by pburkholder** [comment]
Thanks for the comments @hillaryj and @bengerman13. I'm merging as is for now, but I have more tricks I want to add so I'll act on your comments on the next iteration.

**Pull request merged**